ui/src/app/base/components/GroupCheckbox/GroupCheckbox.tsx

```diff
@@ -6,31 +6,38 @@ import { nanoid } from "@reduxjs/toolkit";
 import classNames from "classnames";
 
 import { someInArray, someNotAll } from "app/utils";
+import type { CheckboxHandlers } from "app/utils/generateCheckboxHandlers";
 
-type Props<R, S> = {
+type Props<S> = {
+  checkSelected?: CheckboxHandlers<S>["checkSelected"] | null;
   disabled?: boolean;
-  handleGroupCheckbox: (rows: R[], selected: S[]) => void;
+  handleGroupCheckbox: CheckboxHandlers<S>["handleGroupCheckbox"];
   inRow?: boolean;
-  items: R[];
+  items: S[];
   // This needs to be something other than `label` to prevent conflicts with the
   // HTMLInputElement type.
   inputLabel?: ReactNode;
   selectedItems: S[];
 } & HTMLProps<HTMLInputElement>;
 
-const GroupCheckbox = <R, S>({
+const GroupCheckbox = <S,>({
```

Comment:
Should this just be `<S>` now?

Reply:
That's a funny case when using typescript generics inside a jsx file, the convention is to add a comma otherwise it tries to parse it as an HTML tag.